uxarray/grid/coordinates.py

```diff
@@ -207,6 +141,25 @@ def _populate_lonlat_coord(grid):
     )
 
 
+def _populate_node_xyz(grid) -> None:
+    """Populates the Cartesiain node coordinates of a Grid (`node_x`, `node_y`
+    and `node_z`)"""
+
+    node_lon_rad = np.deg2rad(grid.node_lon.values)
+    node_lat_rad = np.deg2rad(grid.node_lat.values)
+    x, y, z = _lonlat_rad_to_xyz(node_lon_rad, node_lat_rad)
```

Comment:
@erogluorhan 

Relating to your comment, I've used the vectorized implementation when populating coordinates. This will lead to significant performance increases when constructing coordinates that were not present in the original grid.

Reply:
Yes, this is what I meant for the test cases. Thanks!

Reply:
Relatedly, do we have a lot of other places in our (non-test) code, which is not in the scope of #730 either, that we will need to switch to vectorized call?

Reply:
> Relatedly, do we have a lot of other places in our (non-test) code, which is not in the scope of #730 either, that we will need to switch to vectorized call?

No, there only places we'd need to update the calls to be vectorized are
* Zonal average helper functions (`integrate.py`)
* Face area helper functions (`area.py`)
* Arc helper functions (`arc.py`)